src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -372,81 +557,497 @@ public static void sleep(long millis, int nanos)
+     * otherwise be <a href="Thread.html#inheritance">inherited</a>. Once set, a
+     * {@code Thread} or {@code ThreadFactory} is created with the following methods:
+     *
+     * <ul>
+     *     <li> The {@linkplain #unstarted(Runnable) unstarted} method creates a new
+     *          <em>unstarted</em> {@code Thread} to run a task. The {@code Thread}'s
+     *          {@link Thread#start() start} method must be invoked to schedule the
+     *          thread to execute.
+     *     <li> The {@linkplain #start(Runnable) start} method creates a new {@code
+     *          Thread} to run a task and schedules the thread to execute.
+     *     <li> The {@linkplain #factory() factory} method creates a {@code ThreadFactory}.
+     * </ul>
+     *
+     * <p> A {@code Thread.Builder} is not thread safe. The {@code ThreadFactory}
+     * returned by the builder's {@code factory()} method is thread safe.
+     *
+     * <p> Unless otherwise specified, passing a null argument to a method in
+     * this interface causes a {@code NullPointerException} to be thrown.
+     *
+     * @see Thread#ofPlatform()
+     * @see Thread#ofVirtual()
+     * @since 19
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.VIRTUAL_THREADS)
+    public sealed interface Builder
+            permits Builder.OfPlatform,
+                    Builder.OfVirtual,
+                    ThreadBuilders.BaseThreadBuilder {
+
+
+        /**
+         * Sets the thread name.
+         * @param name thread name
+         * @return this builder
+         */
+        Builder name(String name);
+
+        /**
+         * Sets the thread name to be the concatenation of a string prefix and
+         * the string representation of a counter value. The counter's initial
+         * value is {@code start}. It is incremented after a {@code Thread} is
+         * created with this builder so that the next thread is named with
+         * the new counter value. A {@code ThreadFactory} created with this
+         * builder is seeded with the current value of the counter. The {@code
+         * ThreadFactory} increments its copy of the counter after {@link
+         * ThreadFactory#newThread(Runnable) newThread} is used to create a
+         * {@code Thread}.
+         *
+         * @apiNote
+         * The following example creates a builder that is invoked twice to start
+         * two threads named "{@code worker-0}" and "{@code worker-1}".
+         * {@snippet :
+         *   Thread.Builder builder = Thread.ofPlatform().name("worker-", 0);
+         *   Thread t1 = builder.start(task1);   // name "worker-0"
+         *   Thread t2 = builder.start(task2);   // name "worker-1"
+         * }
+         *
+         * @param prefix thread name prefix
+         * @param start the starting value of the counter
+         * @return this builder
+         * @throws IllegalArgumentException if count is negative
```

Comment:
```suggestion
         * @throws IllegalArgumentException if start is negative
```

Reply:
Thanks, I think that dates from when the param was renamed.